File: backend/internal/ws/message.go
package ws

// WsMessage WebSocket 消息结构体
// 定义了前后端通信的数据格式 (JSON)
type WsMessage struct {
	SenderID        uint   `json:"senderId"`                  // 发送者 ID
	ReceiverID      uint   `json:"receiverId"`                // 接收者 ID (0 表示广播/群聊)
	Content         string `json:"content"`                   // 消息内容
	Type            string `json:"type"`                      // 消息类型: text, image, etc.
	MediaURL        string `json:"mediaUrl"`                  // 媒体文件地址 (可选)
	ClientMessageID string `json:"clientMessageId,omitempty"` // 客户端临时消息 ID（用于乐观更新回执）
	MessageID       uint   `json:"messageId,omitempty"`       // 服务端落库后的消息 ID
}

File: IMChat/IMChat/Models/Message.swift
import Foundation

/// 消息模型
/// 用于解析 WebSocket 消息和历史记录 API 响应
struct Message: Codable, Identifiable, Equatable {
    // 使用 UUID 生成唯一 ID，因为 SwiftUI List 需要 Identifiable
    // 如果后端没有返回唯一的消息 ID，我们在本地生成一个
    var id: String = UUID().uuidString
    
    let content: String
    let senderId: Int
    let receiverId: Int
    let type: String // "text", "image", "read_ack"
    let mediaUrl: String?
    let createdAt: String // Go 后端的时间字符串
    var isRead: Bool?
    
    // 映射 JSON 键名
    // 根据前端 stores/chat.ts，后端返回的字段是 PascalCase (大驼峰)
    enum CodingKeys: String, CodingKey {
        case content = "Content"
        case senderId = "SenderID"
        case receiverId = "ReceiverID"
        case type = "Type"
        case mediaUrl = "MediaURL"
        case createdAt = "CreatedAt"
        case isRead = "IsRead"
    }
}
